python/pip_install/private/generate_whl_library_build_bazel.bzl

```diff
@@ -154,28 +176,37 @@ def generate_whl_library_build_bazel(
         if item not in _data_exclude:
             _data_exclude.append(item)
 
+    # Ensure this list is normalized
+    group_deps = [normalize_name(d) for d in group_deps]
+
     lib_dependencies = [
-        "@" + repo_prefix + normalize_name(d) + "//:" + _PY_LIBRARY_LABEL
+        "@" + repo_prefix + normalize_name(d) + "//:" + PY_LIBRARY_PUBLIC_LABEL
         for d in dependencies
+        if d not in group_deps
     ]
     whl_file_deps = [
-        "@" + repo_prefix + normalize_name(d) + "//:" + _WHEEL_FILE_LABEL
+        "@" + repo_prefix + normalize_name(d) + "//:" + WHEEL_FILE_PUBLIC_LABEL
         for d in dependencies
+        if d not in group_deps
     ]
```

Comment:
nit: having `non_group_deps = [d for d in dependencies if d not in group_deps]` and then using that to generated `lib_dependencies` and `whl_file_deps` would be great!